src/plugins/testsBundle/index.js

```diff
@@ -18,21 +19,34 @@ module.exports = (kibana) => {
         let modules = [];
         let config = kibana.config;
 
-        // add the modules from all of the apps
-        for (let app of apps) {
-          modules = union(modules, app.getModules());
-        }
+        const testGlobs = ['src/ui/public/**/*.js'];
+        const testingPluginId = config.get('testsBundle.pluginId');
+
+        if (testingPluginId) {
+          const plugin = plugins.byId[testingPluginId];
+
+          // add the modules from all of this plugins apps
+          for (let app of plugin.apps) {
+            modules = union(modules, app.getModules());
+          }
 
-        const testGlobs = [
-          'src/ui/public/**/__tests__/**/*.js',
-        ];
+          testGlobs.push(
+            '!src/ui/public/**/__tests__/**/*',
+            `${plugin.publicDir}/**/__tests__/**/*.js`
+          );
+        } else {
 
-        for (const plugin of plugins) {
-          testGlobs.push(`${plugin.publicDir}/**/__tests__/**/*.js`);
+          // add the modules from all of the apps
+          for (let app of apps) {
+            modules = union(modules, app.getModules());
+          }
+
+          for (const plugin of plugins) {
```

Comment:
I know you didn't just introduce this here, but can you change this to `let`?  This is currently only working because it's transpiled away from being an actual `const`, but if it were being processed according to spec, this would fail silently outside of strict mode and throw an error inside of strict mode.